packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/common_services/push_notifications_service.dart

```diff
@@ -0,0 +1,62 @@
+{{> licence.dart }}
+
+import 'package:permission_handler/permission_handler.dart';
+
+import '../../assets.dart';
+import '../models/errors/error_model.dart';
+import '../repositories/push_notification_repository.dart';
+
+class PushNotificationsService {
+  final PushNotificationRepository _pushNotificationRepository;
+
+  PushNotificationsService(
+    this._pushNotificationRepository,
+  );
+
+  Future<void> unsubscribe([bool setNotifications = false]) async {
+    final token = await _pushNotificationRepository.getToken();
+    if (token == null) {
+      throw ErrorModel();
+    }
+    await _pushNotificationRepository.unsubscribe(token);
+    await _pushNotificationRepository
+        .setNotificationSubscribed(setNotifications);
+  }
+
+  Future<bool> areNotificationsEnabled() async {
+    final permissionStatus = await Permission.notification.status;
+    final shouldSubscribe =
+        await _pushNotificationRepository.notificationsSubscribed();
+    if (permissionStatus == PermissionStatus.granted && shouldSubscribe) {
```

Comment:
I think for enums like `PermissionStatus` it's a good practice to use `switch` and list all the cases explicitly. This way if there's any change to the statuses in the future, we'll get compile-time feedback.